crates/vm/levm/src/opcode_handlers/stack_memory_storage_flow.rs

```diff
@@ -177,21 +177,36 @@ impl VM {
 
         let address = current_call_frame.to;
 
-        let original_value = if self.cache.is_slot_cached(&address, key) {
-            self.cache
-                .get_storage_slot(address, key)
-                .expect("Storage slot should have been cached")
-                .original_value
+        let mut base_dynamic_gas: U256 = U256::zero();
+
+        let storage_slot = if self.cache.is_slot_cached(&address, key) {
+            self.cache.get_storage_slot(address, key).unwrap()
```

Comment:
Do not unwrap, handle this.

Reply:
Same as https://github.com/lambdaclass/lambda_ethereum_rust/pull/1029/files#r1829787920